application/backend/app/api/endpoints/datasets.py

```diff
@@ -272,3 +277,31 @@ def delete_dataset_item_annotation(
         dataset_service.delete_dataset_item_annotations(project_id=project_id, dataset_item_id=dataset_item_id)
     except ResourceNotFoundError as e:
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=str(e))
+
+
+@router.patch(
+    "/{dataset_item_id}/subset",
+    status_code=status.HTTP_200_OK,
+    responses={
+        status.HTTP_200_OK: {"description": "Dataset item subset is assigned"},
+        status.HTTP_400_BAD_REQUEST: {"description": "Invalid dataset item ID or project ID"},
+        status.HTTP_404_NOT_FOUND: {"description": "Dataset item or project not found"},
+        status.HTTP_422_UNPROCESSABLE_ENTITY: {"description": "Invalid subset"},
+        status.HTTP_409_CONFLICT: {"description": "Dataset item already has a subset assigned"},
```

Comment:
```suggestion
        status.HTTP_409_CONFLICT: {"description": "Dataset item already has a subset assigned"},
        status.HTTP_422_UNPROCESSABLE_ENTITY: {"description": "Invalid subset"},
```
Please sort the responses by code